services/QuillLMS/app/services/google_integration/classroom/creators/students.rb

```diff
@@ -34,49 +34,71 @@ def self.get_student_data_for_all_classrooms(classrooms, students_requester_and_
   end
 
   def self.create_students(student_data)
-    students = student_data.map do |k, v|
-      create_student(student_data[k])
-    end
-    students.compact
+    student_data
+      .map { |k, v| create_student(student_data[k]) }
+      .compact
   end
 
   def self.create_student(data, counter=0)
-    if counter > 2
-      return nil
+    return nil if data[:email].nil?
+    return nil if counter > 2
+
+    update_existing_student_with_google_id_and_different_email(data)
+
+    student = User.find_or_initialize_by(email: data[:email].downcase)
+
+    if student.new_record?
+      student.update(
+        name: data[:name],
+        role: 'student',
+        password: data[:last_name],
+        username: generate_username(data),
+        signed_up_with_google: true
+      )
+    else
+      student.update(
+        account_type: 'Google Classroom',
+        google_id: data[:google_id]
+      )
     end
-    if data[:email]
-      student = User.find_or_initialize_by(email: data[:email].downcase)
-      if student.new_record?
-        classroom = Classroom.unscoped.find(data[:classrooms].first)
-        username = GenerateUsername.new(
-          data[:first_name],
-          data[:last_name],
-          classroom.code
-        ).call
-        student.update(name: data[:name],
-                       role: 'student',
-                       password: data[:last_name],
-                       username: username,
-                       signed_up_with_google: true,
-                       account_type: 'Google Classroom',
-                       google_id: data[:google_id]
-                      )
-      else
-        student.update(account_type: 'Google Classroom', google_id: data[:google_id])
-      end
 
-      if student.errors.any?
-        student = create_student(data, counter += 1)
-      else
-        data[:classrooms].each do |id|
-          classroom = Classroom.find(id)
-          Associators::StudentsToClassrooms.run(student, classroom)
-        end
-      end
-      student
+    if student.errors.any?
+      student = create_student(data, counter += 1)
```

Comment:
We don't have to change this since it's just a duplicate of what the code was doing before, but this doesn't need to use `+=` to increment `counter` since it's passing the value, and the locally scoped value isn't referenced again.  It's a little more legible to just pass `counter + 1` instead, but up to you if you want to bother.

Reply:
That's a good point.  I'll probably just leave as if for now and rewrite during a later refactor.